Extension/src/LanguageServer/extension.ts

```diff
@@ -199,6 +218,60 @@ function onInterval(): void {
     clients.ActiveClient.onInterval();
 }
 
+async function installVsix(vsixLocation: string, updateChannel: string): Promise<void> {
+    // Get the path to the VSCode command -- replace logic later when VSCode allows calling of
+    // workbench.extensions.action.installVSIX from TypeScript w/o instead popping up a file dialog
+    return PlatformInformation.GetPlatformInformation().then((platformInfo) => {
+        const vsCodeScriptPath: string = function(platformInfo): string {
+            if (platformInfo.platform === 'win32') {
+                const vsCodeProcessPath: string = path.dirname(process.execPath);
+                const scriptName: string = path.basename(process.execPath) + '.cmd';
+                return path.join(vsCodeProcessPath, 'bin', scriptName);
+            } else {
+                const vsCodeProcessPath: string = path.basename(process.execPath);
+                const stdout: Buffer = execSync("which " + vsCodeProcessPath);
+                return stdout.toString().trim();
+            }
+        }(platformInfo);
+        if (!vsCodeScriptPath) {
+            telemetry.logLanguageServerEvent('findVsCodeScriptFailure');
```

Comment:
Rather than have a bunch of different telemetry event names, I think there should be one event with a result status payload.  We can communicate the error type via the rejected promise.  So for this case we should remove the telemetry call and reject the promise with a `new Error("findVSCodeScript")` object. In the final catch or error handler, we take the string from the error object and log it with the `'installVsix'` event using `error` or `success` as the payload.